rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,149 @@
+etc).
+
+## Interface
+The `setupDeps.screenshotMode` object has a single purpose: tell the app if it
+should render in an optimized way for screenshot capture:
+
+```
+interface IScreenshotModeServiceSetup {
+  isScreenshotMode: () => boolean;
+}
+```
+
+Internally, this object is constructed from a class that refers to information
+sent via a custom proprietary header:
+
+```
+interface HeaderData {
+  'X-Screenshot-Mode': true
+}
+
+class ScreenshotModeServiceSetup implements IScreenshotModeServiceSetup {
+  constructor(rawData: HeaderData) {}
+  public isScreenshotMode (): boolean {}
+}
+```
+
+This works because the headless browser that opens the page can inject custom
+headers into the request. Teams can test how their app renders when loaded with
+this header using a new configuration setting, or a web debugging proxy, or
+some other tool that is TBD.
+
+# Alternatives
+
+- Print media query CSS
+  If applications UIs supported printability using `@media print`, and Kibana 
+  Reporting uses `page.print()` to capture the PDF, it would be easy for application 
+  developers to test, and prevent bugs showing up in the report.
+  
+  However, this solution doesn't include performance benefits of reducing objects 
+  in the headless browser memory: the headless browser still has to render the entire 
+  page as a "normal" render before it is able to call `page.print()`. No one sees the 
+  results of that initial render, so it is the same amount of wasted rendering cycles 
+  during report generation that we have today.
+
+# Adoption strategy
+
+The Reporting Services team should create an few example in a developer example plugin
+on how to integrate an App with the Screenshot Mode Service. From there, the team should 
+work with App teams in a consulting role to establish usage of this service.
+
+# How we teach this
+
+The Reporting Services team can offer statistics in a weekly update about how many 
+Reporting-enabled applications are using this service. This will help people remember
+that it is available. If the team can also provide statistics about how many bugs this is 
+fixing, how much time it saves in generating a report, etc, then it will also help
+people understand why it is important.
+
+# Further examples
+
+- Applications can also use screenshot context to customize the way they load.
+  An example is Toast Notifications: by default they auto-dismiss themselves
+  after 30 seconds or so. That makes sense when there is a human there to
+  notice the message, read it and remember it. But if the page is loaded for
+  capturing a screenshot, the toast notifications should never disappear. The
+  message in the toast needs to be part of the screenshot for its message to
+  mean anything, so it should not force the screenshot capture tool to race
+  against the toast timeout window.
+- Avoid collection and sending of telemetry from the browser when page is
+  loaded for screenshot capture.
```

Comment:
I might be misremembering, but don't we already have a dedicated header that could be injected for this exact purpose?

Reply:
We don't - you may be remembering the first draft of this RFC that was closed :)